internal/model/analysis.go

```diff
@@ -123,6 +124,9 @@ type PairwiseAnalysisSummary struct {
 	AggregateTotalMeanSimilarities util.AggregateValues            `json:"aggregate-total-mean-similarity"`
 }
 
+// A named map type allows for custom marshalling and unmarshalling.
+type PairwiseAnalysisMap map[PairwiseKey]*PairwiseAnalysis
```

Comment:
Since we need the marshaling for this now (even with the below comment),
make a real quick test for it (just marshal and back).